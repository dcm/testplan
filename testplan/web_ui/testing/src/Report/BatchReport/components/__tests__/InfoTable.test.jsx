/** @jest-environment jsdom */
// @ts-nocheck
import React from 'react';
import { render } from '@testing-library/react';
import { StyleSheetTestUtils } from 'aphrodite';
import { InfoTable } from '../';
import useReportState from '../../hooks/useReportState';
import { filterObjectDeep } from '../../../../__tests__/testUtils';
import { TESTPLAN_REPORT_2 } from '../../../../__tests__/documents';

jest.mock('../../hooks/useReportState');

const TESTPLAN_REPORT_2_SLIM = filterObjectDeep(
  TESTPLAN_REPORT_2,
  [ 'information', 'timer', 'run', 'start', 'end' ]
);

describe('InfoTable', () => {

  beforeEach(() => {
    useReportState.mockName('useReportState');
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it('renders as expected', () => {
    useReportState.mockReturnValue([ TESTPLAN_REPORT_2_SLIM ]);
    const { container } = render(<InfoTable/>);
    expect(container).toMatchSnapshot();
  });

  it('grabs correct slices from useReportState', () => {
    const expectedHookArgs = [ 'app.reports.batch.jsonReport', false ];
    useReportState.mockReturnValue([ TESTPLAN_REPORT_2_SLIM ]);
    render(<InfoTable/>);
    expect(useReportState).toHaveBeenCalledTimes(1);
    expect(useReportState).toHaveBeenLastCalledWith(...expectedHookArgs);
  });

  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    jest.resetAllMocks();
  });

});
