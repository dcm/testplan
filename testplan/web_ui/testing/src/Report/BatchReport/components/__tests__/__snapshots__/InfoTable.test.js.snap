// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTable renders as expected 1`] = `
<div>
  <div
    class="table-responsive"
  >
    <table
      class="infoTable_bqrygy table table-bordered"
    >
      <tbody>
        <tr>
          <td
            class="infoTableKey_1ki1om5"
          >
            user
          </td>
          <td
            class="infoTableValue_x5c1q5"
          >
            unknown
          </td>
        </tr>
        <tr>
          <td
            class="infoTableKey_1ki1om5"
          >
            command_line_string
          </td>
          <td
            class="infoTableValue_x5c1q5"
          >
            /home/unknown/path_to_testplan_script/testplan.py
          </td>
        </tr>
        <tr>
          <td>
            start
          </td>
          <td>
            2018-10-15T14:30:10.998071+00:00
          </td>
        </tr>
        <tr>
          <td>
            end
          </td>
          <td>
            2018-10-15T14:30:11.296158+00:00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
